packages/cli/generation/remote-generation/remote-workspace-runner/src/publishDocs.ts

```diff
@@ -169,6 +169,35 @@ export async function publishDocs({
                 definitionV2: undefined
             });
 
+            if (response.ok) {
+                context.logger.debug(`Registered API Definition ${response.body.apiDefinitionId}`);
+                return response.body.apiDefinitionId;
+            } else {
+                switch (response.error.error) {
+                    case "UnauthorizedError":
+                    case "UserNotInOrgError": {
+                        return context.failAndThrow(
+                            "You do not have permissions to register the docs. Reach out to support@buildwithfern.com"
+                        );
+                    }
+                    default:
+                        if (apiName != null) {
+                            return context.failAndThrow(`Failed to register API ${apiName}`, response.error);
+                        } else {
+                            return context.failAndThrow("Failed to register API", response.error);
+                        }
```

Comment:
@RohinBhargava nit: users won't really know what it means to register an API -- it might be helpful to have a more clear error message

`Failed to publish docs because API definition (${apiName}) could not be uploaded. Please contact support@buildwithfern.com\n ${response.error}`